bentoml/frameworks/spark_mllib.py

```diff
@@ -0,0 +1,128 @@
+import importlib
+import json
+import logging
+import os
+
+from bentoml.exceptions import BentoMLException, MissingDependencyException
+from bentoml.service import BentoServiceArtifact, BentoServiceEnv
+
+
+class PySparkModelArtifact(BentoServiceArtifact):
+    """
+    PySparkModelArtifact allows you to use the spark.ml and spark.mllib APIs with Bento
+
+    Example usage:
+
+    >>> from pyspark.sql import SparkSession
+    >>>
+    >>> spark = SparkSession.builder.getOrCreate()
+    >>>
+    >>> # Load training data
+    >>> training = spark.read.format("libsvm").load("data/mllib/sample_libsvm_data.txt")
+
+    >>> lr = LogisticRegression(maxIter=10, regParam=0.3, elasticNetParam=0.8)
+
+    >>> # Fit the model
+    >>> lrModel = lr.fit(training)
+    >>>
+    >>> import bentoml
+    >>> from bentoml.frameworks.sklearn import PySparkModelArtifact
+    >>> from bentoml.adapters import DataframeInput
+    >>>
+    >>> @bentoml.env(infer_pip_packages=True)
+    >>> @bentoml.artifacts([PySparkModelArtifact('model')])
+    >>> class SparkModelService(bentoml.BentoService):
+    >>>
+    >>>     @bentoml.api(input=DataframeInput(), batch=True)
+    >>>     def predict(self, df):
+    >>>         result = self.artifacts.model.predict(df)
+    >>>         return result
+    >>>
+    >>> svc = SparkModelService()
+    >>>
+    >>> # Pack directly with sklearn model object
+    >>> svc.pack('model', model_to_save)
+    """
+
+    def __init__(self, name: str, spark=None):
```

Comment:
What's the reason for this `spark` parameter here?

Reply:
The `__init__` parameter is mostly used for the configuration of a model artifact, ideally, something that could be easily serialized to a YAML file so we could recreate the artifact instance with the given config, and load it from a serialized files.  

I think it is ok if we don't support loading a Spark model with a user-provided spark session for now, since we are focusing on the online serving use case. For batch inference on Spark, we might need to add additional APIs at the artifact level, which we could leave for another PR/discussion, open to suggestions!

Reply:
Interesting, do you think I should just remove all references to `self.spark` in that case? I agree, it likely wouldn't be used particularly often.